src/headers/tomcrypt_cipher.h

```diff
@@ -1031,6 +1031,21 @@ int salsa20_test(void);
 
 #endif /* LTC_SALSA20 */
 
+#ifdef LTC_XSALSA20
+
+int xsalsa20_setup(salsa20_state *st, const unsigned char *key,   unsigned long keylen,
+                                      const unsigned char *nonce, unsigned long noncelen,
+                                      int rounds, unsigned char *subkey);
```

Comment:
shouldn't there be a `unsigned long *subkeylen`? :)

Reply:
added